make/tests

```diff
@@ -1,67 +1,57 @@
 N_TESTS ?= 100 # override this in make/local. If <= 0, N_TESTS + 1 is interpreted as the number of batches to group the tests into
 
 ##
-# LIBGTEST is the google test library
 # GTEST_MAIN is the file that contains the google test
 ##
-LIBGTEST = test/libgtest.a
 GTEST_MAIN = $(GTEST)/src/gtest_main.cc
-CFLAGS_GTEST += -isystem $(GTEST)/include -isystem $(GTEST)
+GTEST_CXXFLAGS += -isystem $(GTEST)/include -isystem $(GTEST) -O$O
 
 ##
 # Build the google test library.
 ##
-$(LIBGTEST): $(LIBGTEST)(test/gtest.o)
+$(GTEST)/src/gtest-all.o: CXXFLAGS += $(GTEST_CXXFLAGS)
 
-test/gtest.o: $(GTEST)/src/gtest-all.cc
-	@mkdir -p test
-	$(COMPILE.cc) -O$O $(CFLAGS_GTEST) $< $(OUTPUT_OPTION)
+test/%$(EXE) : CXXFLAGS += $(GTEST_CXXFLAGS)
```

Comment:
Yes, it would use that `CXXFLAGS` for the things it would need to build as part of its dependencies. For details, see the part of the [GNU make manual](https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html) that describes this.